[changelog] add possibility to overwrite previous tag Resolves #382

andreabettich opened the pull request:
Resolves #382 

### Context
<!-- What problem does this change address? -->
<!-- Link to relevant issues or discussions here -->

### Checklist
- [x] [Review Contribution Guidelines](https://github.com/jreleaser/jreleaser/blob/master/CONTRIBUTING.md).
- [x] Make sure all contributed code can be distributed under the terms of the 
      [Apache License 2.0](https://github.com/jreleaser/jreleaser/blob/master/LICENSE), e.g. the code was written by 
      you or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Update [documentation](https://github.com/jreleaser/jreleaser.github.io) when applicable.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/jreleaser/jreleaser?pullRequest=383) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/jreleaser/jreleaser?pullRequest=383) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/jreleaser/jreleaser?pullRequest=383) it.</sub>

aalmiray commented on sdks/git-sdk/src/main/java/org/jreleaser/sdk/git/ChangelogGenerator.java:
```diff
@@ -238,10 +239,19 @@ private static Comparable version(JReleaserContext context, Ref tag, Pattern ver
+        String previousTagName = gitService.getConfiguredPreviousTagName();
+        Optional<Ref> previousTag;
+
+        if(StringUtils.isNotBlank(previousTagName)){
```
Please preserve white space as every other `if` statement does

aalmiray commented on plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/internal/dsl/AbstractGitService.groovy:
```diff
@@ -56,6 +56,7 @@ abstract class AbstractGitService implements GitService {
     final Property<String> username
     final Property<String> token
     final Property<String> tagName
+    final Property<String> previousTagName
```
Please add a `Property<String> getPreviousTagName()` to the `GitService.groovy` DSL interface.

aalmiray commented:
Thank you @andreabettich. You found most places to make the required changes, only missing updates for the Maven plugin. The following types must be updated as well:

https://github.com/jreleaser/jreleaser/blob/ba1ed9b50d2921789b9b95cdb4b583321d1a5504/plugins/jreleaser-maven-plugin/src/main/java/org/jreleaser/maven/plugin/GitService.java
https://github.com/jreleaser/jreleaser/blob/ba1ed9b50d2921789b9b95cdb4b583321d1a5504/plugins/jreleaser-maven-plugin/src/main/java/org/jreleaser/maven/plugin/JReleaserAutoConfigReleaseMojo.java
https://github.com/jreleaser/jreleaser/blob/ba1ed9b50d2921789b9b95cdb4b583321d1a5504/plugins/jreleaser-maven-plugin/src/main/java/org/jreleaser/maven/plugin/internal/JReleaserModelConverter.java

andreabettich commented:
I fixed the formatting and added the missing parts. Sorry for already amending and force-pushing onto the branch. makes it less transparent what I changed.

aalmiray commented:
Thank you. Force push is OK. I'll do one full review again and then we'll be good to go. 

If you feel like doing so, you could also send a PR to jreleaser/jreleaser.github.io for updating configuration snippets. If not that's OK, I can do that later 😄

andreabettich commented:
Added some documentation https://github.com/jreleaser/jreleaser.github.io/pull/16

aalmiray commented:
Thank you for your contribution! 🎉 
You don't have to wait til `v0.8.0` is released to test out this feature. You can grab the latest `early-access` release and invoke the CLI directly. Alternatively the jreleaser/jreleaser-slim:early-access Docket image should work with GitLab CI as well

https://hub.docker.com/r/jreleaser/jreleaser-slim/tags

Just have tom wait til the current early-access build finishes to grab the binaries.

andreabettich commented:
awesome, thanks a lot!
